work_queue/src/work_queue_worker.c

```diff
@@ -603,7 +603,7 @@ static int handle_tasks(struct link *master)
 				char *sandbox_name = string_format("%s/%s",p->sandbox,f->remote_name);
 
 				debug(D_WQ,"moving output file from %s to %s",sandbox_name,f->payload);
-				if(rename(sandbox_name,f->payload)!=0) {
+				if(copy_file_to_file(sandbox_name,f->payload)!=0) {
```

Comment:
Careful here: If rename works, we want to use it, since it is much cheaper.  So, try the rename, and if that fails with errno==EXDEV, then use copy_file_to_file.